### ISSUE TITLE
[1.7.1] Relogin after reboot

### ISSUE THREAD

**Issue by cdamken** [owncloud/client#2736]
### Expected behaviour
Both windows and mac 1.7.1. sync client will keep the session after reboot.

### Actual behaviour
Both windows and mac 1.7.1. sync client will loose it's session after reboot. not always, but most often a user needs to relogin

### Steps to reproduce
1. Reboot the computer
2. Start the client
3. Relogin (Should not ask for it)

### Server configuration
CentOS

Apache

Galera Cluster

Shibboleth

ownCloud version: 6.0.4

Updated from older ownCloud

```
### Logs
01-22 12:37:49:295 0x7fe8eaf283e0 networkjobs.cpp:210 !!! Mirall::CheckServerJob created for  QUrl( "https://SERVERNAME/files" )  querying "status.php"
01-22 12:37:49:336 0x7fe8eaf283e0 networkjobs.cpp:456 status.php returns:  QMap(("edition", QVariant(QString, "enterprise") ) ( "installed" ,  QVariant(QString, "true") ) ( "version" ,  QVariant(QString, "6.0.4.1") ) ( "versionstring" ,  QVariant(QString, "6.0.4") ) )    0  Reply:  QNetworkReplyHttpImpl(0x7fe8eaf75c20)
01-22 12:37:49:337 0x7fe8eaf283e0 connectionvalidator.cpp:99 ** Application: ownCloud found:   QUrl( "https://SERVERNAME/files/status.php" )   with version  "6.0.4" ( "6.0.4.1" )
01-22 12:37:49:347 0x7fe8eaf283e0 networkjobs.cpp:210 !!! Mirall::PropfindJob created for  QUrl( "https://SERVERNAME/files" )  querying "/"
01-22 12:37:49:347 0x7fe8eaf283e0 connectionvalidator.cpp:156 # checking for authentication settings.
01-22 12:37:49:361 0x7fe8eaf283e0 shibbolethcredentials.cpp:187 void Mirall::ShibbolethCredentials::slotReplyFinished(QNetworkReply *) detected redirect, will open Login Window
01-22 12:37:49:361 0x7fe8eaf283e0 networkjobs.cpp:554 Quota request *not* successful, http result code is 302 "https://AUTHSERVER/authentication/idp/single-sign-on/key:KEYYYYYYYYYY"
01-22 12:37:49:363 0x7fe8eaf283e0 owncloudgui.cpp:264 Folder in overallStatus Message:  Mirall::Folder(0x7fe8eafd83d0)  with name  "carlos"
01-22 12:37:49:365 0x7fe8eaf283e0 owncloudgui.cpp:264 Folder in overallStatus Message:  Mirall::Folder(0x7fe8eafd83d0)  with name  "carlos"
01-22 12:37:50:705 0x7fe8eaf283e0 folderman.cpp:559 FolderMan: Syncing is disabled, no scheduling.
01-22 12:37:52:126 0x7fe8eaf283e0 sparkleupdater_mac.mm:107 virtual void Mirall::SparkleUpdater::backgroundCheckForUpdate() launching background check
01-22 12:37:52:137 0x7fe8eaf283e0 sparkleupdater_mac.mm:34 -[DelegateObject updaterMayCheckForUpdates:] may check: YES
01-22 12:37:53:110 0x7fe8eaf283e0 sparkleupdater_mac.mm:47 -[DelegateObject updaterDidNotFindUpdate:]
```
@MorrisJobke 

00002447

**Comment by cdamken** [comment]
HA-Proxy 1.5 with session=cookies is in front of the oC6.0.4

**Comment by cdamken** [comment]
It looks that the session of shibboleth is the only one is saved on the HA-Proxy, but not the sync-client session. Is there any way to add it on the synclient?

**Comment by efef** [comment]
Almost; the HAproxy adds a session cookie to create a sticky session, but when the client is closed it only stores the shibboleth session cookie, which is useless if you don't have the proxy cookie.
The next time the client comes online it may end up at a different application server due to the missing session cookie from HAproxy. And the new application server doesn't know the shibboleth session id and will redirect the user to re-login again.

**Comment by guruz** [comment]
So this is not really a client bug is it? The cookie lifetime of the HA-Proxy needs to be higher?

@danimo @ogoffart We store _all_ cookies right?

**Comment by danimo** [comment]
We store the shib cookie separately in the keychain, because it's a credential that is usually not supposed to be stored (and only would have session lifetime). All other cookies are preserved and restored (you can see that e.g. because the WayF keeps the preferred IdP).

**Comment by cdamken** [comment]
@guruz @danimo I wrote the information in this text (I can't post it because of the private information)

S3-ownCloud\Shared\owncloud\support\github-issues\mirall\2736\Answer.txt

**Comment by dragotin** [comment]
I think the problem here is that the request to .../status.php sets a new cookie. That must not be, as the call to status.php happen unauthorized. @danimo?

**Comment by danimo** [comment]
The analysis I've seen is accurate: the client does not store cookies for accesses of `status.php`. Is there any way the load balancer can be kept from tagging accesses to status.php? subsquent tagging should work fine.

The long term fix is to implement cookie storage thoroughly, see https://github.com/owncloud/client/issues/2808

**Comment by zijlstra-it** [comment]
@danimo I assume @cdamken has put my analyses in the "Answer.txt" file but for clearance, my analyses should show you that you indeed store all the right cookies and use them but not in a particular situation: the fresh start of the client.

The client does a request to "status.php" and gets a new cookie from the proxy; no problem here because status.php is unprotected by Shibboleth.

The problem then starts with the client is requesting an update on the users file directory for the first time after the start; it then uses the new cookies instead of the one saved in the "cookies.db" file and a login using shibboleth is triggered.

When the login has been done and a new shibboleth session has been created the client then starts to request all the directory updates using the OLD cookies, which are still valid.

So basically the user gets a new login but eventually still uses the old session.

**Comment by dragotin** [comment]
scheduling this for the first bugfix release of 1.8 line.

**Comment by cdamken** [comment]
Shibboleth provides a new cookie because the session is send to the wrong backend because the client uses the new cookie when doing a PROPFIND for the first time rather then the stored cookies, or else the session would have ended at the correct backend where the shibboleth sessions known.

We fixed it by using memcache as session store so all backends see the same sessions.

Still it's a client issue in my view.

**Comment by dragotin** [comment]
Ok, happy that its fixed for this specific case. We will revisit that when we work on 1.8.1.

**Comment by craigpg** [comment]
@cdamken, can we close this?  No more troubleshooting steps that I can see.

**Comment by craigpg** [comment]
Trouble shooting is over.  Changing label.

**Comment by cdamken** [comment]
@craigpg Yes we can close this one when @dragotin  
>We will revisit that when we work on 1.8.1.
